### ISSUE TITLE
'onprocessfiles' doesnt seem to trigger, file.status always 2

### ISSUE THREAD

**Issue by Suedo** [pqina/react-filepond#137]
### Summary
[Codesandbox demo of issue](https://codesandbox.io/s/filepond-react-status-check-2zlre?file=/src/ImgUploadComponent.tsx:1671-1685)

Using the latest versions of Filepond and Filepond-React. Trying to disable/enable an upload button based on status of files being added, but the `onprocessfiles` callback is not firing.

```ts
    <FilePond
      ref={pondref}
      files={files}
      allowMultiple={true}
      onupdatefiles={filesUpdated}
      maxParallelUploads={1}
      acceptedFileTypes={["image/png", "image/jpeg"]}
      onprocessfiles={() => {
        console.log("all done");
      }}
      fileValidateTypeLabelExpectedTypesMap={{
        "image/jpeg": ".jpg",
        "image/png": ".png"
      }}
      labelIdle={disp}
    />
```

Checked the docs, onprocessfiles takes no args:
![image](https://user-images.githubusercontent.com/4963779/84798295-6812fd00-b018-11ea-81ce-a882077c322e.png)

As an alternative, tried using the individual file statuses, but all of them seem to come as 2 (IDLE) always:

```ts
  useEffect(() => {
    const status = images.some((i) => i.status != 5); // Filepond.Filestatus of 2 == idle
    console.log('checking upload status', images.map((i) => i.status).join(','));
    setUploading((uploading) => status);
  }, [images]); // array of image files
```

### How to reproduce

Codesandbox [here](https://codesandbox.io/s/filepond-react-status-check-2zlre?file=/src/ImgUploadComponent.tsx:1671-1685) (same link as above)

### Expected behaviour

`onprocessfiles` will fire , and/or file.status will change

Raised this issue as #81 never saw any good resolution.

**Comment by rikschennink** [comment]
For configurations without the `server` property `onprocessfiles` is not called, it's only called when files are processed.

**Comment by Suedo** [comment]
Ok. what about the `file.status` not changing, and only staying at 2?

In the sandbox, if you take a look at the console, it only shows file.status as 2. I uploaded 3 images, and my callbak fired 3 times, but with the same status. 

![image](https://user-images.githubusercontent.com/4963779/84812824-a2869500-b02c-11ea-9b70-4a3ac7e01cd7.png)

**Comment by rikschennink** [comment]
I don’t see any logic for uploading the files? So they’re idling.

**Comment by Suedo** [comment]
I am adding a file from my local ... how to see the status then, if the file(s) has completed loading successfully ?

**Comment by rikschennink** [comment]
If status is idle the file has loaded

**Comment by Suedo** [comment]
But the base64 url of the file isn't available immediately, when using file-encode plugin, as Mentioned earlier [here](https://github.com/pqina/filepond-plugin-file-encode/issues/13#issuecomment-643952611)

That is why I have to wait for a while. Once it shows in the browser that the files are uploaded, trying to get the base64 url's works.

**Comment by rikschennink** [comment]
https://github.com/pqina/filepond-plugin-file-encode/issues/13#issuecomment-643966746

**Comment by Suedo** [comment]
Per your [last comment in the file-encoder#13](https://github.com/pqina/filepond-plugin-file-encode/issues/13#issuecomment-644278093) when using `onpreparefile` to add files, one can use onremovefile to remove a file. 

I tried that one out, **has different issue: the remove callback fires multiple times during fileadd**, and causes error. Codesandbox [here](https://codesandbox.io/s/filepond-react-status-check-2zlre?file=/src/ImgUploadComponent.tsx:1882-1894)

That is why I went with the current approach of using `onupdatefiles` , which led to a different set of issues, which I reported here. 

Closed this issue as your replies aren't really helping out. I'll stick to using the ugly timeout hack.
